sgbatch/src/computation/CacheComputation.h

```diff
@@ -0,0 +1,37 @@
+
+
+#ifndef S_CACHECOMPUTATION_H
+#define S_CACHECOMPUTATION_H
+
+#include <wrench-dev.h>
+
+#include "../SimpleSimulator.h"
+
+class CacheComputation {
+
+public:
+    CacheComputation(std::set<std::shared_ptr<wrench::StorageService>> & storage_services,
+                std::vector<std::shared_ptr<wrench::DataFile>> &files,
+                double total_flops
+    );
+
+    void determineFileSources(std::string hostname);
+
+    void operator () (std::shared_ptr<wrench::ActionExecutor> action_executor);
+
+    double determineFlops(double data_size, double total_data_size);
+
+    virtual void performComputation(std::string &hostname);
+
+protected:
```

Comment:
Must be protected. Otherwise, the child class objects won't have access. Is there a more elegant way?

Reply:
no, that's exactly what protected is for